### PR TITLE
Firebase 2way communication

### PR DESCRIPTION
**Pull request by aravind-segu**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
 
   return (
-    <div align="center">
-      <LoginComponent />
-      <MainGameScreen />
-    </div>
+    <Router>
+      <div>
+        {/* A <Switch> looks through its children <Route>s and
+          renders the first one that matches the current URL. */}
```
Comment (Shehan29) [inline src/App.js:30]: Woah that’s cool

File: src/backend/startup.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+var playerID = "";
+
+export async function register(name) {
+    await firestore.collection(roomName).add({
+        name: name,
+    })
+    .then(function(docRef) {
+        playerID = docRef.id;
+        console.log("Document written with ID: ", playerID);
+    })
```
Comment (Shehan29) [inline src/backend/startup.js:14]: Since you defined a .then is this function still async?

**Review by Shehan29 — APPROVED** [review]

File: src/backend/startup.js (hunk 2 above)
Comment (ShivamDh) [reply src/backend/startup.js:14]: I think `register` is meant to be an exported async function that's made up of async + sync(callback) logic.

**Comment by ShivamDh** [comment]
Let me know if the popups are working as desired

**Pull request merged**